pandas/core/internals.py

```diff
@@ -139,14 +139,14 @@ def is_categorical_astype(self, dtype):
         validate that we have a astypeable to categorical,
         returns a boolean if we are a categorical
         """
-        if is_categorical_dtype(dtype):
-            if dtype == CategoricalDtype():
-                return True
-
+        if dtype is Categorical or dtype is CategoricalDtype:
```

Comment:
hmm, this seems to contract the

``elif is_categorical_dtype(dtype)``

IOW we DO allow ``CategoricalDtype``, just not ``Categorical/CategoricalIndex`` here

you can just use ``pandas_dtype`` I think

Reply:
Some test was failing without this earlier. I'll see if I can revert it and figure out what was going on.

Reply:
Ah, this is to disallow passing in the class `CategoricalDtype`, instead of an instance. Without the check, you get an object dtype when (incorrectly) passing `CategoricalDtype` (not an instance)

```python
In [17]: pd.Series([1, 2]).astype(CategoricalDtype)
Out[17]:
0    1
1    2
dtype: object
```